module/documents/actor/actor.mjs

```diff
@@ -753,6 +753,22 @@ export default class Actor5e extends Actor {
     return allowed !== false ? this.update(updates, {dhp: -amount}) : this;
   }
 
+  /* -------------------------------------------- */
+ 
+  /**
+   * Apply a certain amount of temporary hit point, but only if it's more than the actor currently has.
+   * @param {number} amount       An amount of temporary hit points to set
+   * @returns {Promise<Actor5e>}  A Promise which resolves once the temp HP has been applied
+   */
+  async applyTempHP(amount=0) {
```

Comment:
It seems this function is specifically intended for cases where the character may **gain** temporary HP. As such I think perhaps we should explicitly name it `gainTemporaryHP` and reject cases where `amount < 0`, or do I misunderstand?

Reply:
If you pass in a negative number, then `amount > tmp` will be false since it would be something like `-5 > 0` and it won't update anything.